FirebaseAuth/Sources/Auth/FIRAuth.m

```diff
@@ -1304,6 +1388,75 @@ - (void)sendPasswordResetWithNullableActionCodeSettings:
         passwordResetRequestWithEmail:email
                    actionCodeSettings:actionCodeSettings
                  requestConfiguration:self->_requestConfiguration];
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
```

Comment:
Same comment here about visionOS and anywhere else iOS only needs to be ifdef'd.

Reply:
Could you elaborate a bit? I don't fully understand "iOS only needs to be ifdef'd".

Reply:
Because `TARGET_OS_IOS` evaluates as true when building for Vision OS.

Reply:
Yeah, it has been applied to this PR.

There are many existing usage of `TARGET_OS_IOS`. I'm thinking make a separate pr with a custom micro like `TARGET_OS_IOS_EXCLUDE_MACCATALYST_VISONOS`, which should make the code simpler. WDYT?

Reply:
SGTM, Crashlytics does something like that